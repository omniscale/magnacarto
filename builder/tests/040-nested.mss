#roads{
  [type='rail'] {
    [service='yard'] {
      line-color: red;
    }
  }
}

#roads[zoom=17] {
  line-width: 2;
  line-color: yellow;

  [type='rail'] {
    line-width: 5;
    // service='yard' should be red
  }
}

#roads {
  line-width: 1;
}