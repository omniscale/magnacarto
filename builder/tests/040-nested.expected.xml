<Map srs="epsg:3857">
  <Parameters></Parameters>
  <Style name="roads" filter-mode="first">
    <Rule>
      <!--Zoom{=17}-->
      <MaxScaleDenominator>5000</MaxScaleDenominator>
      <MinScaleDenominator>2500</MinScaleDenominator>
      <Filter>(([service] = &#39;yard&#39;) and ([type] = &#39;rail&#39;))</Filter>
      <LineSymbolizer stroke="#ff0000" stroke-width="5"></LineSymbolizer>
    </Rule>
    <Rule>
      <Filter>(([service] = &#39;yard&#39;) and ([type] = &#39;rail&#39;))</Filter>
      <LineSymbolizer stroke="#ff0000" stroke-width="1"></LineSymbolizer>
    </Rule>
    <Rule>
      <!--Zoom{=17}-->
      <MaxScaleDenominator>5000</MaxScaleDenominator>
      <MinScaleDenominator>2500</MinScaleDenominator>
      <Filter>([type] = &#39;rail&#39;)</Filter>
      <LineSymbolizer stroke="#ffff00" stroke-width="5"></LineSymbolizer>
    </Rule>
    <Rule>
      <!--Zoom{=17}-->
      <MaxScaleDenominator>5000</MaxScaleDenominator>
      <MinScaleDenominator>2500</MinScaleDenominator>
      <LineSymbolizer stroke="#ffff00" stroke-width="2"></LineSymbolizer>
    </Rule>
    <Rule>
      <LineSymbolizer stroke-width="1"></LineSymbolizer>
    </Rule>
  </Style>
  <Layer name="roads" srs="" status="off">
    <StyleName>roads</StyleName>
  </Layer>
</Map>